UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs

```diff
@@ -167,6 +173,14 @@ public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenACATALOGFi
                     bool checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(folderPath, $"B{baseNumber}", Config.ExchangeSetEncRootFolder), Config.ExchangeSetCatalogueFile);
                     Assert.IsTrue(checkFile, $"{Config.ExchangeSetCatalogueFile} File not Exist in the specified folder path : {Path.Combine(folderPath, Config.ExchangeSetEncRootFolder)}");
```

Comment:
Hi Raymond, We don't have any endpoint to check the file exists or not. That's y we are using loop to traverse each base folder to check the file exists or not. Thanks

Reply:
Then will it be quick to confirm file exists then test and validate the contents or the format afterwards in that way we con confirm that the files are present and if not then fail. i would be good to validate what files are expected and confirm each in order to see which ones are missing out of the list
we could probable think about [testcasesource] as a directive

Reply:
Hi Raymond, That's exactly what we are doing but at the individual folder level. so that specific test can be re-used to run in isolation as well.
As per the POS folder structure for large media files, the CATALOG.031 is expected in each Base folder(i.e B1-B9). Hence, we are first validating the file exist in the folder , if it is TRUE then we go forward and test the file contents.

Below code helps us to validate the expected file exists/not.

 bool checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(folderPath, $"B{baseNumber}", Config.ExchangeSetEncRootFolder), Config.ExchangeSetCatalogueFile);
 Assert.IsTrue(checkFile, $"{Config.ExchangeSetCatalogueFile} File not Exist in the specified folder path : {Path.Combine(folderPath, Config.ExchangeSetEncRootFolder)}");